src/osemgrep/cli_ci/Ci_subcommand.ml

```diff
@@ -179,8 +179,16 @@ let at_url_maybe ppf () : unit =
 let decode_json_rules caps (data : string) : Rule_fetching.rules_and_origin =
   Common2.with_tmp_file ~str:data ~ext:"json" (fun file ->
       let file = Fpath.v file in
-      Rule_fetching.load_rules_from_file ~rewrite_rule_ids:false ~origin:App
-        ~registry_caching:false caps file)
+      match
+        Rule_fetching.load_rules_from_file ~rewrite_rule_ids:false ~origin:App
+          ~registry_caching:false caps file
+      with
+      | Left rules -> rules
+      | Right _err ->
+          (* There shouldn't be any errors, because we obtained these rules
```

Comment:
this will probably happen :) better use a failwith with a good error message (your comment), than
a Common.Impossible.